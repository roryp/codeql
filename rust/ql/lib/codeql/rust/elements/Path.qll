// generated by codegen, do not edit
/**
 * This module provides the public class `Path`.
 */

private import internal.PathImpl
import codeql.rust.elements.PathSegment
import codeql.rust.elements.Resolvable

/**
 * A path. For example:
 * ```rust
 * foo::bar;
 * ```
 */
final class Path = Impl::Path;
