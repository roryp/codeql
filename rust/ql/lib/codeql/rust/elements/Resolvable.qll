// generated by codegen, do not edit
/**
 * This module provides the public class `Resolvable`.
 */

private import internal.ResolvableImpl
import codeql.rust.elements.AstNode

/**
 * Either a `Path`, or a `MethodCallExpr`.
 */
final class Resolvable = Impl::Resolvable;
