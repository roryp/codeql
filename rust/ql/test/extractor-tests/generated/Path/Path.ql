// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

from
  Path x, string hasResolvedPath, string hasResolvedCrateOrigin, string hasQualifier, string hasPart
where
  toBeTested(x) and
  not x.isUnknown() and
  (if x.hasResolvedPath() then hasResolvedPath = "yes" else hasResolvedPath = "no") and
  (
    if x.hasResolvedCrateOrigin()
    then hasResolvedCrateOrigin = "yes"
    else hasResolvedCrateOrigin = "no"
  ) and
  (if x.hasQualifier() then hasQualifier = "yes" else hasQualifier = "no") and
  if x.hasPart() then hasPart = "yes" else hasPart = "no"
select x, "hasResolvedPath:", hasResolvedPath, "hasResolvedCrateOrigin:", hasResolvedCrateOrigin,
  "hasQualifier:", hasQualifier, "hasPart:", hasPart
