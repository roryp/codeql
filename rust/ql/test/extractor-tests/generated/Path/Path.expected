| gen_path.rs:5:5:5:7 | foo | hasResolvedPath: | no | hasResolvedCrateOrigin: | no | hasQualifier: | no | hasPart: | yes |
| gen_path.rs:5:5:5:12 | foo::bar | hasResolvedPath: | no | hasResolvedCrateOrigin: | no | hasQualifier: | yes | hasPart: | yes |
